K2Bridge/Factories/AggregateFactory.cs

```diff
@@ -0,0 +1,230 @@
+                {
+                    var key = column.ColumnName;
+                    aggregateDictionary.Add(key, GetValueAggregate(key, row, logger));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Get percentile aggregate from a given <see cref="DataRow"/>.
+        /// </summary>
+        /// <param name="columnName">The column name.</param>
+        /// <param name="columnMetadata">The column metadata.</param>
+        /// <param name="row">The row to be parsed.</param>
+        /// <param name="logger">ILogger object for logging.</param>
+        /// <returns><see cref="PercentileAggregate"></returns>
+        private static PercentileAggregate GetPercentileAggregate(string columnName, string[] columnMetadata, DataRow row, ILogger logger)
+        {
+            logger.LogTrace("Get percentile aggregate for {}", columnName);
+
+            // Parse list of percents, and keyed option
+            var percents = columnMetadata[2..^1];
+            var keyed = bool.Parse(columnMetadata[^1]);
+
+            var percentileAggregate = new PercentileAggregate() { Keyed = keyed };
+
+            if (row[columnName] is JArray percentileValues && percentileValues.HasValues)
+            {
+                foreach (var (percent, value) in percents.Zip(percentileValues))
+                {
+                    var percentileItem = value.Type switch
+                    {
+                        // If token type is a string, we assume this is is date time
+                        JTokenType.String => new PercentileItem()
+                        {
+                            Percentile = double.Parse(percent, CultureInfo.InvariantCulture),
+                            Value = TimeUtils.ToEpochMilliseconds(value.Value<DateTime>()),
+                            ValueAsString = value.Value<DateTime>().ToString("yyyy-MM-ddTHH:mm:ss.fffK"),
+                        },
+                        _ => new PercentileItem()
+                        {
+                            Percentile = double.Parse(percent, CultureInfo.InvariantCulture),
+                            Value = value.Value<double>(),
+                        },
+                    };
+
+                    percentileAggregate.Values.Add(percentileItem);
+                }
+            }
+            else
+            {
+                // If row[columnName] is empty, it returns null value for each percentile requested
+                foreach (var percent in percents)
+                {
+                    var percentileItem = new PercentileItem()
+                    {
+                        Percentile = double.Parse(percent, CultureInfo.InvariantCulture),
+                        Value = null,
+                    };
+
+                    percentileAggregate.Values.Add(percentileItem);
+                }
+            }
+
+            logger.LogTrace("Percentile aggregate returned for {0}: {1}", columnName, percentileAggregate);
```

Comment:
```suggestion
            logger.LogTrace("Percentile aggregate returned for {}: {}", columnName, percentileAggregate);
```
(And in the other one too)

Reply:
Fixed.